### PR TITLE
fix(search): keep play counts out of artist metadata

### PR DESCRIPTION
**Pull request by LUC4N3X**
## Summary

Fix YouTube Music search metadata parsing so popularity metrics such as play/view/stream counts are not exposed as artist or album labels. Artist references are resolved across renderer columns and packed artist/album metadata is handled correctly.

## What changed

- ignore view/play/stream counts, years and durations when selecting search-result artist and album labels
- prefer structured YouTube Music artist references even when metrics shift metadata into later renderer columns
- preserve valid album labels and fall back to `YouTube Music` instead of exporting popularity counts as metadata
- add focused regression coverage for the reported `27M plays` case and packed artist/album columns

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Performance improvement
- [ ] Refactor or maintenance
- [ ] UI or UX change
- [ ] Localization or accessibility
- [ ] Build, CI, packaging, or release change
- [ ] Documentation
- [ ] Breaking change

## Scope

- **Platform:** Android
- **Area:** Streaming / YouTube Music metadata parsing

### Automated checks

- [ ] Android: `./gradlew --no-daemon :app:lintRelease :app:testReleaseUnitTest :app:assembleRelease`
- [ ] Desktop: `cd desktop && ./gradlew check assemble`
- [x] Targeted tests were added or updated
- [x] Not applicable, with the reason explained below

The original focused search-renderer tests and adjacent YouTube Music parser tests passed before the CodeRabbit follow-up commit. CodeRabbit's sandbox could not rerun the selected Gradle tests after the follow-up because its configured `JAVA_HOME` points to a missing JDK directory. GitHub PR checks are the source of truth for the final branch state.

### Manual testing

| Environment | Scenario | Result |
|---|---|---|
| N/A | Parser-only change; covered with focused renderer fixtures | N/A |

## Risk and compatibility

- **Risk level:** Low
- **Compatibility or migration notes:** None
- **Known limitations:** YouTube Music renderer shapes can change; focused regression fixtures cover the known cases in this PR
- **Rollback plan:** Revert this PR

## Reviewer notes

- Main parser change: `YoutubeMusicRepository.kt`
- Regression coverage: `YoutubeMusicSearchRendererTest.kt`
- No version, dependency, permission, workflow, database, F-Droid metadata, or release-channel changes

## Release note

Fixed incorrect artist and album metadata appearing in some YouTube Music search results.

## Related issues

N/A

## Checklist

- [x] The PR is focused and contains no unrelated changes
- [x] The implementation follows the existing architecture and naming conventions
- [x] Threading, lifecycle, cancellation, and resource cleanup were reviewed where relevant
- [x] Tests, documentation, localization, and screenshots were updated where required
- [x] No secrets, keystores, generated packages, archives, or local-only files were committed
- [x] Android and Desktop versioning and release channels remain independent
- [ ] CI is green, or every remaining failure is explained in this PR

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved artist and album detection across varied search result layouts.
  - Prevented play counts, localized metrics, and metadata labels from being mistaken for artists or albums.
  - Preserved structured album and artist references, including collaborating artists and browse information.
  - Improved exact-artist searches by replacing missing or placeholder artist details with verified information.
  - Album searches now use enriched track results for more accurate results.

### CODE REVIEW

**Comment by LUC4N3X** [comment]
@coderabbitai autofix

**Comment by LUC4N3X** [comment]
@coderabbitai Please make the following additional targeted fixes in this PR and commit them directly to the current branch `codex/fix-search-result-metadata`. Do not open a new PR.

Verify each item against the current code first. Keep the patch minimal and limited to YouTube Music metadata parsing plus focused regression tests.

1. `parseMusicRenderer`: the current `artistTokenIndex` fallback can still export a collaborator / multi-artist label as the album. Prefer a real YouTube Music album browse reference when the renderer provides one (reuse `extractYoutubeMusicAlbumReference`), and propagate its browse ID into `Track.albumBrowseId`. For the text fallback, skip metadata that represents the resolved artist or the resolved artist-reference group, so a second artist or combined artist label is not selected as the album. Preserve valid album labels and the existing packed-column fix.

2. `parseCarouselItem` for `musicTwoRowItemRenderer`: use the same plausible-metadata filtering and artist-reference precedence used by `parseMusicRenderer`. Right now its first subtitle token can still be a play/view count or other metadata instead of the artist. Preserve `artistBrowseIds`.

3. `isYoutubeMusicAlbumTrackMetadata`: reject localized play/view/stream metrics consistently with `isPlausibleYoutubeMusicAlbumTitle` (for example `27 M reproducciones`, and the existing localized metric fragments already recognized there). Prefer one shared metric-word helper/constant instead of two different rules.

4. Add focused regressions for at least: packed artist+album still works; multiple/collaborating artists are not selected as album; an album browse endpoint wins and its browse ID is preserved; a localized metric such as `27 M reproducciones` is ignored; and the two-row carousel artist resolution if it can be covered cleanly without broad unrelated refactors.

Do not change versions, dependencies, workflows, permissions, database, release metadata, or unrelated UI. Run the focused tests and adjacent YouTube Music parser tests available in the repo, and report any check that cannot run.

**Comment by LUC4N3X** [comment]
@coderabbitai The invalid `JAVA_HOME` is an environment issue in your sandbox, not a reason to leave the confirmed parser defects unfixed. Please continue with the code changes already requested, commit them directly to the current PR branch, and do not open a stacked PR. Run any static/diff checks you can; if Gradle still cannot run because of that environment issue, report it clearly but still deliver the minimal code + regression-test changes.

**Comment by LUC4N3X** [comment]
@coderabbitai full review

Please explicitly verify these current-head edge cases as part of the full review:
- `parseMusicRenderer` can select a collaborator/second artist as `album` after the first resolved artist; prefer a structured album browse endpoint when available and preserve `albumBrowseId`.
- `parseCarouselItem` for `musicTwoRowItemRenderer` still takes the first non-type subtitle token as artist, so a play/view metric can leak into artist metadata instead of preferring artist references.
- `isYoutubeMusicAlbumTrackMetadata` recognizes fewer localized metric words than `isPlausibleYoutubeMusicAlbumTitle`; e.g. `27 M reproducciones` can pass through search metadata filtering.

Please post actionable review findings only for issues that are valid on the current head, with focused regression-test guidance.

**Comment by LUC4N3X** [comment]
@coderabbitai full review

Please explicitly verify these current-head edge cases:
1. `parseMusicRenderer` may select a collaborator/second artist as `album`; prefer a structured album browse endpoint when available and preserve `albumBrowseId`.
2. `parseCarouselItem` for `musicTwoRowItemRenderer` may still expose a play/view metric as artist instead of preferring structured artist references.
3. `isYoutubeMusicAlbumTrackMetadata` has narrower localized metric detection than `isPlausibleYoutubeMusicAlbumTitle`; for example `27 M reproducciones` can leak into search metadata.

Post actionable findings only for issues valid on the current head, including focused regression-test guidance.

File: app/src/main/java/com/luc4n3x/levyra/data/YoutubeMusicRepository.kt [hunk 1]
```diff
@@ -2221,16 +2223,28 @@ class YoutubeMusicRepository(private val context: Context? = null) {
...
+            ?.takeIf(String::isNotBlank)
+            ?: fallbackArtist
+        val album = subtitleLines
+            .drop(1)
+            .flatMap { it.split(" • ", " · ", " - ") }
+            .map(String::trim)
+            .firstOrNull { token ->
+                isPlausibleSearchMetadataLabel(token) && !token.equals(artist, ignoreCase = true)
+            }
             ?: "YouTube Music"
```
[the remark this answers is not in the document]
Comment (LUC4N3X) [reply app/src/main/java/com/luc4n3x/levyra/data/YoutubeMusicRepository.kt]: Additional still-valid edge cases found while verifying the Autofix. Please include these in the next fix pass; keep the patch limited to `YoutubeMusicRepository.kt` and focused regression tests.

<details>
<summary>🤖 Prompt for AI Agents</summary>

```
Verify each item against current head and apply only valid fixes.

1. In parseMusicRenderer, do not infer album by simply taking the first plausible token after the first resolved artist. With collaborations/multiple artist runs, that can export the second artist or combined artist label as album. Prefer extractYoutubeMusicAlbumReference(renderer) when it returns a real album name/browse ID, propagate that browse ID to Track.albumBrowseId, and make the text fallback exclude the resolved artist-reference group as album metadata.

2. In parseCarouselItem for musicTwoRowItemRenderer, resolve fallback artist using isPlausibleSearchMetadataLabel and then prefer extractYoutubeMusicArtistReferences(two, fallbackArtist).firstOrNull()?.name. This prevents play/view/stream metadata from becoming the artist and preserves artistBrowseIds.

3. Make localized metric filtering consistent between isYoutubeMusicAlbumTrackMetadata and isPlausibleYoutubeMusicAlbumTitle. Reuse one metric-word helper/list so values such as `27 M reproducciones` are rejected by search metadata parsing too.

4. Add focused regression tests for: collaborator/second artist not becoming album; structured album endpoint winning and albumBrowseId being preserved; localized `27 M reproducciones` ignored; and the two-row carousel metric/artist-reference shape if it can be tested cleanly. Preserve the existing packed-column regression.

Do not touch versions, dependencies, workflows, permissions, database, F-Droid metadata, release metadata, or unrelated UI. If the sandbox JAVA_HOME is invalid, still deliver the generated source/test changes and report the verification limitation.
```

</details>

**Comment by LUC4N3X** [comment]
@coderabbitai autofix

File: app/src/main/java/com/luc4n3x/levyra/data/YoutubeMusicRepository.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (LUC4N3X) [reply app/src/main/java/com/luc4n3x/levyra/data/YoutubeMusicRepository.kt]: Resolved on current head `a5e0c071`.

- packed artist/album metadata is parsed from the full subtitle token sequence;
- collaborators and combined artist-reference metadata are excluded from album fallback;
- structured album references win and preserve `albumBrowseId`;
- two-row carousel parsing prefers artist references over metric text and preserves `artistBrowseIds`;
- localized metric detection is shared by album-title and track-metadata filtering;
- regressions cover packed columns, collaborators, album browse IDs, `27 M reproducciones`, and two-row carousel parsing. A direct predicate assertion for the Spanish metric was also added.

Validation performed locally with JDK 21 and Android SDK 36:

- focused and adjacent YouTube Music parser tests: PASS;
- `:app:lintRelease -PlevyraFdroidBuild=true`: PASS;
- `:app:assembleRelease -PlevyraFdroidBuild=true`: PASS;
- agent configuration and AI-efficiency validators: PASS;
- `git diff --check`: PASS.

The complete unit suite ran 653 tests: 649 passed and 4 failed outside this PR diff (two missing local decoder parity resources, one existing localized album-label assertion, and one player-expansion assertion). All tests covering this change pass.

File: app/src/test/java/com/luc4n3x/levyra/viewmodel/ReliableArtistSearchTest.kt [hunk 2]
```diff
@@ -74,6 +76,48 @@ class ReliableArtistSearchTest {
...
+
+        val result = enrichSearchTracksWithExactArtist(
+            query = "felices los 4",
+            results = SearchResults(topTrack = placeholder, songs = listOf(placeholder)),
+            reliableArtists = listOf(artist("Maluma", "UC-maluma"))
+        )
+
+        assertEquals("YouTube Music", result.topTrack?.artist)
+    }
+
```
[the remark this answers is not in the document]
Comment (LUC4N3X) [reply app/src/test/java/com/luc4n3x/levyra/viewmodel/ReliableArtistSearchTest.kt:120]: Addressed in d266ce94. Added a regression test proving that enrichment replaces the placeholder artist name while preserving an existing artistBrowseIds value. The focused ReliableArtistSearchTest suite passes.
